typescript/codepipeline-using-cross-account-repos/index.ts

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+import 'source-map-support/register';
+import cdk = require('@aws-cdk/core');
+import codecommit= require('@aws-cdk/aws-codecommit');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import codepipeline_actions = require('@aws-cdk/aws-codepipeline-actions');
+import iam = require('@aws-cdk/aws-iam');
+import s3 = require('@aws-cdk/aws-s3');
+import kms = require('@aws-cdk/aws-kms');
+ 
+const app = new cdk.App();
+const devEnv = {region: 'us-west-2', account: '123123123123'} 
+const toolEnv = {region: 'us-west-2', account: '776362166164'}
+ 
+// Source Stack
+const srcStack = new cdk.Stack(app, 'srcStack', { env: devEnv });
+ 
+const srcRepo = codecommit.Repository.fromRepositoryArn(srcStack, 'srcRepo', 'arn:aws:codecommit:us-west-2:363575151959:omnisearch')
+ 
+const srcRepoRole = new iam.Role(srcStack, 'CrossAccountRole', {
+  roleName: 'CrossAccountRoleName', 
+  assumedBy: new iam.AccountPrincipal(toolEnv['account']),
+});
+ 
+var srcCrossPolicy = new iam.PolicyStatement();
+srcCrossPolicy.addAllResources();
+srcCrossPolicy.addActions('codecommit:*');
+srcRepoRole.addToPolicy(srcCrossPolicy);
+ 
+//  Pipeline Stack
+const pipeStack = new cdk.Stack(app, 'pipeStack', { env: toolEnv });
+const sourceOutput = new codepipeline.Artifact();
+var pipeline = new codepipeline.Pipeline(pipeStack, 'Pipeline', {
+  artifactBucket: new s3.Bucket(pipeStack, 'Bucket', {
+    bucketName: 'castaren-test-artifact-bucket',
+    encryption: s3.BucketEncryption.KMS,
+    encryptionKey: new kms.Key(pipeStack, 'CrossAccountKmsKey',
+ {
+    removalPolicy: cdk.RemovalPolicy.DESTROY,
+  }
+),
+  }),
+  stages: [
+    {
+      stageName: 'Source',
+      actions: [
+        new codepipeline_actions.CodeCommitSourceAction({
+          actionName: 'Source',
+          repository: srcRepo,
+          output: sourceOutput,
+        }),
+      ],
+    },
+    // pipelines must have a minimum of 2 stages
+    { 
+      stageName: 'Dummyapproval',
+      actions: [
+        new codepipeline_actions.ManualApprovalAction({
+          actionName: 'ManualApproval',
+        }),
+      ],
+    },
+]});
+    
+var pipeCrossPolicy = new iam.PolicyStatement()
+pipeCrossPolicy.addResources(srcRepoRole.roleArn);
```

Comment:
What is `srcRepoRole`? Is not used anywhere...

Reply:
next week works :) 

srcRepoRole is created at the top within the srcStack, and it's used by the pipeStack in order to access the repos cross account

I can run a test to verify that the role is necessary, but I'm pretty sure I needed it in order for the pipeline to function

it gives access

Reply:
You create a Role, create a Policy, attach the Policy to the Role, then attach the Policy to the Pipeline. I also don’t see where the Role is used.

Reply:
This isn’t doing what you think it’s doing. You’re giving the policy permission to do “something” with the Role Arn, not attaching the Policy to the role. I think your naming conventions may be leading to some of the confusion. I would suggest adding a verb to the role names. “PipelineExecutionPolicy”, “SourceReadRole”, etc....